modules/consentManagement.js

```diff
@@ -182,6 +189,101 @@ function lookupIabConsent(cmpSuccess, cmpError, hookConfig) {
   }
 }
 
+function lookupUspConsent(uspSuccess, cmpError, hookConfig) {
+  function handleCmpResponseCallbacks() {
+    const uspResponse = {};
+
+    function afterEach() {
+      if (uspResponse.usPrivacy) {
+        uspSuccess(uspResponse, hookConfig);
+      }
+    }
+
+    return {
+      consentDataCallback: function (consentResponse) {
+        uspResponse.usPrivacy = consentResponse.usPrivacy;
+        afterEach();
+      }
+    }
+  }
+
+  let callbackHandler = handleCmpResponseCallbacks();
+  let uspapiCallbacks = {};
+
+  // the following code also determines where the USP is located and uses the proper workflow to communicate with it:
+  // check to see if USP is found on the same window level as prebid and call it directly if so
+  // check to see if prebid is in a safeframe (with USP support)
+  // else assume prebid may be inside an iframe and use the IAB USP locator code to see if USP's located in a higher parent window. this works in cross domain iframes
+  // if the USP is not found, the iframe function will call the cmpError exit callback to abort the rest of the USP workflow
+  let f = window;
+  let uspFrame;
+  while (!uspFrame) {
+    try {
+      if (f.frames['__uspapiLocator']) uspFrame = f;
+    } catch (e) { }
+    if (f === window.top) break;
+    f = f.parent;
+  }
+
+  if (!uspFrame) {
+    return cmpError('USP not found.', hookConfig);
+  }
+
+  callUspWhileInIframe('getUSPData', uspFrame, callbackHandler.consentDataCallback);
```

Comment:
what happened to the direct `__uspapi()` call?  if it's available we should call the function directly so we don't have to wait for a `postMessage`

Reply:
this would work I think:
```
try {
    uspFrame.__uspapi('getUSPData', 1, callbackHandler.consentDataCallback);
} catch () {
    callUspWhileInIframe('getUSPData', uspFrame, callbackHandler.consentDataCallback);
}
```
(check my syntax of course)